pkg/engine/context/loaders/globalcontext.go

```diff
@@ -0,0 +1,138 @@
+package loaders
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/go-logr/logr"
+	kyvernov1 "github.com/kyverno/kyverno/api/kyverno/v1"
+	enginecontext "github.com/kyverno/kyverno/pkg/engine/context"
+	"github.com/kyverno/kyverno/pkg/engine/jmespath"
+	"github.com/kyverno/kyverno/pkg/engine/variables"
+	globalcontextstore "github.com/kyverno/kyverno/pkg/globalcontext/store"
+)
+
+type gctxLoader struct {
+	ctx       context.Context //nolint:containedctx
+	logger    logr.Logger
+	entry     kyvernov1.ContextEntry
+	enginectx enginecontext.Interface
+	jp        jmespath.Interface
+	gctxStore globalcontextstore.Store
+	data      []byte
+}
+
+func NewGCTXLoader(
+	ctx context.Context,
+	logger logr.Logger,
+	entry kyvernov1.ContextEntry,
+	enginectx enginecontext.Interface,
+	jp jmespath.Interface,
+	gctxStore globalcontextstore.Store,
+) enginecontext.Loader {
+	return &gctxLoader{
+		ctx:       ctx,
+		logger:    logger,
+		entry:     entry,
+		enginectx: enginectx,
+		jp:        jp,
+		gctxStore: gctxStore,
```

Comment:
We don't need a store here, the loader should not depend on the store directly, introduce inversion of dependency :)